src/composables/balances.ts

```diff
@@ -80,12 +82,15 @@ export function useBalances() {
     );
   }
 
-  onNetworkChange(() => {
-    updateBalances();
-  });
-
   initPollingWatcher(() => updateBalances());
 
+  if (!initialized) {
```

Comment:
Were we initiating several `updateBalances` on `onNetworkChange` if the balances we used in several places on particular page previously?

Reply:
No, because previous logic was working in a different way. Every time the composable was fired the logic behind `onNetworkChange` was comparing the previously stored network with the actual and if the values was different the callback was fired. New logic is based on hooks - you add callback to registry and the networks composable is responsible for firing the callbacks.

Thanks to this the composable is actually reacting to network change when it happens instead only when the composable is requested by component.

New logic requires to add the hook callback only once - this is why we needed to have the `initialized` variable here.